### DISCUSSION TITLE
New Operators Concept

### DISCUSSION THREAD

**Discussion by WithSJ** [CsqLang/Csq3#21] [Ideas]
### 1. Arithmetic Operators

- ( + ) Addition

    1 + 2 + 3  | +(1,2,3)

    ```
    var a = 1+2+3

    #function of +
    var a = +(1,2,3)

    Output: 6
    ```

- ( / ) Division

    1 / 2 / 3  |  /(1,2,3)

    ```
    var a = 1/2/3

    #function of /
    var a = /(1,2,3)

    Output: 0.16666666
    ```

**The same thing happened for:-  (-) Subtraction, (*) Multiplication, (/) Division, (%) Modulus, etc.**

### 2. Logical Operators

- (and) Logical AND
    
    True and False and True | and(True,False,True)

    ```
    var a = True and False and True
    
    #function of and
    var a = and(True,False,True)
    
    Output: False
    ```

- (or) Logical OR
    
    True or False or True | or(True,False,True)

    ```
    var a = True orFalse or True
    
    #function of or
    var a = or(True,False,True)
    
    Output: True
    ```

How it will help us
### "+" operator

we can use as a sum function

+(Array List of Numbers)

 it will return the Sum of all value

### "and" Logical AND
```
if condition and condition and condition and condition and condition 
    Statements
end if
```

```
if and(condition,condition ,condition ,condition, condition )
    Statements
end if
```
it's more readable 

## Note
**Right Now it's just an overview of all Operators. try to implement it in the separate git branch of the project**

**Comment by AniketKumar2500** [comment]
Yes your idea looks good so we don't need to do much just have to change Modules/headers/comb.h

**Reply by WithSJ** [reply]
try to implement this.. 👍

**Reply by AniketKumar2500** [reply]
Yeah

**Reply by AniketKumar2500** [reply]
but the condition like:
if cond1 and cond2 looks readable so my idea is to make a function` __and__ `which will do the work similar to you idea of and function what do you say?

**Reply by WithSJ** [reply]
if it is possible to use "and" for both things that are better 
reason for one thing I need to remember two types of __ and __ ,  "and"

**Reply by AniketKumar2500** [reply]
lets try

**Reply by AniketKumar2500** [reply]
But the idea for arithmetic operators looks confusing so lets don't evalute it for them.

**Reply by WithSJ** [reply]
some points I keep in my mind while designing C² Language 
Maybe we can call it our language principles

## Principles of C² language

- One keyword for many syntaxes
- No need for unnecessary things
- keep it simple but significant

**Reply by WithSJ** [reply]
> But the idea for arithmetic operators looks confusing so lets don't evalute it for them.

I didn't do any new thing. actually, it's already implemented in Julia Language
only for +,*,/ but for C² language it will be for all operators

**Reply by AniketKumar2500** [reply]
I tried all of them they are not possible because they are conflicting the IR code.

**Reply by WithSJ** [reply]
> I tried all of them they are not possible because they are conflicting the IR code.

Okay that's why i ask it's possible or not 😄
